kai-client-ci/report/visualizer.js

```diff
@@ -0,0 +1,500 @@
+let KAIData = []
+let JsonDates = []
+
+const CHART_COLORS = {
+    // Colors for Line Elements (Cool and Vivid)
+    blue: 'rgb(54, 162, 235)',
+    green: 'rgb(48, 217, 87)',
+    red: 'rgb(255, 102, 102)',
+    purple: 'rgb(153, 102, 255)',
+    cyan: 'rgb(176, 58, 111)',
+    darkBlue: 'rgb(0, 0, 139)',
+
+    // Colors for Bar Elements (Warm and Muted/Pastel)
+    lightOrange: 'rgb(255, 200, 112)',
+    lightRed: 'rgb(255, 150, 150)',
+    lightBlue: 'rgb(122, 213, 255)',
+    lightBrown: 'rgb(210, 180, 140)',
+    lightPurple: 'rgb(199, 171, 255)',
+
+    // Additional Colors
+    yellow: 'rgb(255, 205, 86)',
+    orange: 'rgb(255, 159, 64)',
+    grey: 'rgb(201, 203, 207)',
+    pink: 'rgb(255, 182, 193)',
+
+    // Transparent versions
+    redTransparent: 'rgba(255, 99, 132, 0.5)',
+    orangeTransparent: 'rgba(255, 159, 64, 0.5)',
+    yellowTransparent: 'rgba(255, 205, 86, 0.5)',
+    greenTransparent: 'rgba(75, 192, 192, 0.5)',
+    blueTransparent: 'rgba(54, 162, 235, 0.5)',
+    purpleTransparent: 'rgba(153, 102, 255, 0.5)',
+    greyTransparent: 'rgba(201, 203, 207, 0.5)',
+    lightBlueTransparent: 'rgb(122, 213, 255, 0.5)',
+    lightPurpleTransparent: 'rgb(199, 171, 255, 0.5)',
+};
+
+// todo: Update to correct path or env variable
+const gistURL = "https://gist.githubusercontent.com/midays/c6e40aac77cbecf8b9a92849bd3393ca/raw/20f1ad8f184704657829249209d1315abf7e7744/newData"
+
+function createDatePicker() {
+    const datePicker = document.getElementById('date-picker');
+
+    const minDate = JsonDates[0];
+    const maxDate = JsonDates[JsonDates.length - 1];
+    datePicker.min = minDate;
+    datePicker.max = maxDate;
+
+    datePicker.addEventListener('change', (event) => {
+        const selectedDate = event.target.value;
+        if (JsonDates.includes(selectedDate)) {
+
+            document.getElementById('pies-chart').remove();
+            const validCodeCanvas = document.createElement('canvas');
+            validCodeCanvas.id = 'pies-chart';
+
+            document.getElementById('average-ranges-chart').remove();
+            const averageRangesCanvas = document.createElement('canvas');
+            averageRangesCanvas.id = 'average-ranges-chart';
+
+            document.getElementById('single-performace-charts').appendChild(validCodeCanvas);
+            document.getElementById('single-performace-charts').appendChild(averageRangesCanvas);
+
+            pieCharts(selectedDate);
+            averageRangesChart(selectedDate);
+        }
+    });
+}
+
+function formatDatesToLabels(dates) {
+
+    const months = ['Jan', 'Feb', 'March', 'April', 'May', 'June', 'July', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec']
```

Comment:
Missing semicolon, please take a look at the whole code to meet the code format PR check

Reply:
done